### PR TITLE
Add attempt count to queue model

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
Add attempt count to queue model to enable retries in async platform.

### CODE REVIEW

File: src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt [hunk 1]
```diff
@@ -27,13 +27,17 @@ import org.veupathdb.lib.rabbit.jobs.serialization.JsonSerializable
 data class ErrorNotification @JvmOverloads constructor(
   val jobID:         HashID,
   val code:          Int,
+  val attemptCount:  Int? = 0,
```
Comment (Foxcapades) [inline src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt]: Why is `attemptCount` nullable here?

File: src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt [hunk 2]
```diff
@@ -46,21 +50,27 @@ data class ErrorNotification @JvmOverloads constructor(
...
         throw IllegalStateException("Error notification has a non-integral ${JsonKey.Code} field!")
 
       return ErrorNotification(
-        HashID(json.get(JsonKey.JobID).textValue()),
-        json.get(JsonKey.Code).intValue(),
-        json.get(JsonKey.Message).textValue()
+        jobID = HashID(json.get(JsonKey.JobID).textValue()),
+        code = json.get(JsonKey.Code).intValue(),
+        message = json.get(JsonKey.Message).textValue(),
+        attemptCount = json.get(JsonKey.AttemptCount).intValue(),
```
Comment (Foxcapades) [inline src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt:78]: With no absence check on `get(JsonKey.AttemptCount)` couldn't this throw an NPE?

AFAIK `ObjectNode.get` will return null if the key is absent and will return a `NullNode` if the key is explicitly set to `null`

**Comment by Foxcapades** [comment]
Two possible code concerns, and a general concern as to whether this belongs as part of the raw dispatch object or as part of the JSON body (where relevant).

The retry count is fine here either way as the async platform is currently the only consumer of this library, but it feels to me like this is a detail specific to the consumers of this project not this project itself?

File: src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt (hunk 2 above)
Comment (dmgaldi) [reply src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/ErrorNotification.kt:78]: Thanks,fixed in new revision

File: src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/JobDispatch.kt [hunk 3]
```diff
@@ -38,13 +37,15 @@ data class JobDispatch(
...
   val type:    String = "",
+  val attemptCount: Int = 0,
   val dispatched: OffsetDateTime = OffsetDateTime.now()
 ) :  JsonSerializable {
   override fun toJson() =
     Json.new<ObjectNode> {
       put(JsonKey.JobID, jobID.string)
       put(JsonKey.Type, type)
       put(JsonKey.Dispatched, dispatched.toString())
+      put(JsonKey.AttemptCount, attemptCount.toString())
```
Comment (Foxcapades) [inline src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/JobDispatch.kt]: Should the `attemptCount` value be stored as a string or int value?

Comment (dmgaldi) [reply src/main/kotlin/org/veupathdb/lib/rabbit/jobs/model/JobDispatch.kt]: Fixed, sorry I got distracted by git-filter-repo yesterday

**Review by Foxcapades — APPROVED** [review]

**Pull request merged**